game/views.py

```diff
@@ -1,10 +1,125 @@
 from django.shortcuts import render
-from django.views.generic import TemplateView
+from django.views import View
+from game.igdb_wrapper import IGDBRequestsHandler
+from game.twitter_wrapper import TwitterWrapper
+from django.core.paginator import Paginator
 
 
-class MainPageView(TemplateView):
-    template_name = 'game/main_page.html'
+class MainPageView(View):
+    """
+    View for main page: displays main page without any filters.
+    Called by initial page entering and by navbar logo click.
+    """
 
+    @staticmethod
+    def get(request):
+        """Gets games from database not filtered"""
 
-class DetailPageView(TemplateView):
-    template_name = 'game/game_detail_page.html'
+        igdb = IGDBRequestsHandler()
+        games = igdb.get_game_main_page_info(18)
+        context = {
+            'page_obj': pagination_generate(request, games, 6),
+            'genres': igdb.get_genres(),
+            'platforms': igdb.get_platforms(),
+        }
+        return render(request, 'game/main_page.html', context)
+
+
+class MainPageViewFilter(View):
+    """
+    View for main page with applied filters: handles requests after 'Apply' button click,
+    displayed in filters sidebar area. Get method handles pagination.
+    """
+
+    def get(self, request):
+        """Processes GET request to main page: displays games from database filtered by pages"""
+
+        igdb = IGDBRequestsHandler()
+        filters = request.session.get('filters')
+        context = self.generate_context(request, igdb, filters)
+        return render(request, 'game/main_page.html', context)
+
+    def post(self, request):
+        """Processes POST request to main page: displays games filtered after filters apply"""
+
+        igdb = IGDBRequestsHandler()
+        filters = {
+            'lower_rating_bound': request.POST.get('rating-begin'),
+            'upper_rating_bound': request.POST.get('rating-end'),
+            'genres': request.POST.getlist('genres'),
+            'platforms': request.POST.getlist('platforms'),
+        }
+
+        request.session['filters'] = filters
+        context = self.generate_context(request, igdb, filters)
+        return render(request, 'game/main_page.html', context)
+
+    @staticmethod
+    def generate_context(request, igdb: IGDBRequestsHandler, filters: dict) -> dict:
```

Comment:
you're over-complicating a bit, just define context as a dict and pass to render, no need to use separate function